nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/AbstractKuduProcessor.java

```diff
@@ -352,6 +351,9 @@ protected void buildPartialRow(Schema schema, PartialRow row, Record record, Lis
                         case VARCHAR:
                             row.addVarchar(colIdx, DataTypeUtils.toString(value, recordFieldName));
                             break;
+                        case DATE:
+                            row.addDate(colIdx, DataTypeUtils.toDate(value, () -> DataTypeUtils.getDateFormat("yyyy-MM-dd"), recordFieldName));
```

Comment:
Maybe use `DataTypeUtils.getDateFormat(RecordFieldType.DATE.getDefaultFormat());` like is done here: https://github.com/apache/nifi/blob/main/nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/util/DataTypeUtils.java#L109

What if the string is in a different format? Do we expect that a user would transform their data? Do other processors handle this?

Reply:
Hive3 processor uses `org.apache.nifi.serialization.record.DataType.getFormat()` and Solr processor uses `RecordFieldType.DATE.getDefaultFormat()`

Reply:
I pushed a fix as a separate commit